Factory.cpp

```diff
@@ -0,0 +1,44 @@
+#include <iostream>
+#include <cmath>
+
+class Point
+{
+private:
+        Point(float x, float y)
+            :m_x(x),
+            m_y(y)
+        {}
+    
+    class PointFactory
+    {
+    public:
+        static Point makePolarPoint(float r, float t)
```

Comment:
Please return a pointer (`std::unique_ptr<Point>`)  instead of an object copy. So you can return nullptr when object creation failed.